models/remoteworkitem_test.go

```diff
@@ -0,0 +1,92 @@
+// +build unit
+
+package models
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"testing"
+)
+
+func TestMap(t *testing.T) {
+
+	// This is the Type to be converted to
+	workItemType := wellKnown["1"] // Will remove this and add an actual Type.
+
+	remoteWorkItem := RemoteWorkItem{
+		Fields: map[string]interface{}{
+			"id":          "3",
+			"description": "I am a remote issue's description in Github",
+			"title":       "I am the title of the issue",
+		},
+	}
+
+	workItemMap := WorkItemMap{
+		"id":          "remote_issue_id",
+		"description": "description",
+		"title":       "title",
+	}
+	t.Log("Test remoteWorkItem", remoteWorkItem)
+	t.Log("Test workItemMap ", workItemMap)
+
+	// TODO: Add error handling
+	workItem := remoteWorkItem.MapRemote(workItemMap, workItemType)
+
+	// TODO: compare the workItem fields with those of the remoteWorkItem
+
+	t.Log("Test workItem ", workItem)
+}
+
+func TestMapGithub(t *testing.T) {
+	url := "https://api.github.com/repos/almighty/almighty-core/issues/131"
```

Comment:
This depends on an external system we can't control which could make our test suite unstable for no good reason.

A good choice here would be to store the content locally and read it from a file instead. That of course, runs the risk of us not testing against real data if the remote system change in some form. 

A good strategy here would be to support 'automatic updatable test data examples'. Example:
https://gist.github.com/aslakknutsen/9ad87dcff9d6c5ad11584112d2deb5d6